kb/2025-11-03/vm-slowness-issue.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Harvester VM Network Slowness – GRO/GSO Offload Issue
+description: 'Reason for VM Network slowness and how to fix it by disabling GRO/GSO offload'
+slug: disable_gro_gso_offload
+authors:
+  - name: Renuka Devi Rajendran
+    title: Senior Software Engineer
+    url: https://github.com/rrajendran17
+    image_url: https://github.com/rrajendran17.png
+tags: [vm network, performance]
+hide_table_of_contents: false
+---
+
+Guest VMs running on a Harvester cluster experienced very slow network throughput across nodes. Disabling Generic Receive Offload (GRO) and Generic Segmentation Offload (GSO) on the Harvester host interfaces resolved the issue.
+
+## Problem
+
+### Symptoms
+
+- Guest VMs exhibited very slow download and transfer speeds (as low as 100 bps).
+- apt-get update, curl, and scp between VMs across nodes were extremely slow.
+- Performance was normal between VMs running on the same node.
+
+### Environment
+
+- Harvester cluster with a mix of Dell and Supermicro nodes.
+- Issue seen on Dell nodes used and not on Supermicro nodes.
+- VM and management traffic shared the same management cluster network.
+- Physical NICs were 100 Gbps interfaces.
+
+## Cause
+
+### Root Cause
+
+Harvester relies on Linux’s bridge-based virtual networking to connect guest VMs to physical networks.
+
+The Dell servers hosting the Harvester cluster were connected to leaf switches that were transmitting Jumbo Frames. These frames should typically be segmented to around ~1450 bytes before reaching the Harvester host kernel when MTU is configured with default value of 1500. However, the packets arriving at the kernel were fragmented into much smaller sizes than expected. This resulted in the kernel having to process a significantly higher number of packets, increasing CPU overhead and causing reduced download throughput.
```

Comment:
any example command, outout of `the packets arriving at the kernel were fragmented into much smaller sizes than expected`